### ISSUE TITLE
Tracking issue for environment/PAM issues

### ISSUE THREAD

**Issue by K900** [nix-community/NixOS-WSL#171]
Upstream issue: https://github.com/microsoft/WSL/issues/9213

Workaround for now: 

```nix
users.users.foo.shell = let
    wrapper = pkgs.writeShellScriptBin "shell-wrapper" ''
      . /etc/set-environment
      exec ${pkgs.zsh}/bin/zsh "$@"
    '';
    in lib.mkForce "${wrapper}/bin/shell-wrapper";
```

Proper workaround doesn't seem possible without nixpkgs changes / infinite recursion.

**Comment by Atry** [comment]
Can we move the workaround to `bashWrapper`?
https://github.com/nix-community/NixOS-WSL/blob/34b95b3962f5b3436d4bae5091d1b2ff7c1eb180/modules/systemd/native/default.nix#L12

**Comment by K900** [comment]
No, that is only used for WSL to run things inside its own `init`. The user shells are spawned directly.
